### PR TITLE
adds aws delete instance, updates instances, vms collection and query class

### PR DESCRIPTION
**Pull request by smatula**
## What does this implement/fix? Explain your changes

- Adds AWS delete instance to the provisioning examples.

- Updates the query methods in the instances and vms collection.
  Added more options to query on. For Instances, Provider, network, tenant, subnet, vendor, type.
  For VMS Provider, vendor, type. 
  Added option --by_id. May execute method by supplying the id of entity instead of name.

- Query class updated so query will return attributes if supplied. Resulting query entities are reloaded
with attributes if passed.

...

## Does this close any currently open issues?

Related to #10 

## More comments

...

**Notes on Subject:**

* succinct description of the change - no more than 60 chars

* use the imperative, present tense: "change" not "changed" nor "changes" nor
  "changing"

* don't capitalize first letter

* no dot (.) at the end

**Notes on body:**

* be descriptive about why this change is useful and how it is an improvement -- the more information the better.

### CODE REVIEW

File: miqcli/query.py [hunk 1]
```diff
@@ -16,10 +16,15 @@
...
 from manageiq_client.api import APIException
 from manageiq_client.filters import Q
-
 from miqcli.utils import log
 
-__all__ = ['BasicQuery', 'AdvancedQuery']
+__all__ = ['BasicQuery', 'AdvancedQuery', 'inject']
+
+
+def inject(lst, item):
```
Comment (vi-patel) [inline miqcli/query.py:24]: Can you add a docstring for inject?

**Review by vi-patel — APPROVED** [review]
looks good, just minor comment about a docstring.

File: miqcli/query.py [hunk 2]
```diff
@@ -73,7 +78,7 @@ def __init__(self, collection):
         """
         super(BasicQuery, self).__init__(collection)
 
-    def __call__(self, query):
+    def __call__(self, query, attr=None):
```
Comment (ryankwilliams) [inline miqcli/query.py:81]: `attr` parameter is missing from the doc string.

File: miqcli/query.py [hunk 3]
```diff
@@ -121,7 +133,7 @@ def __init__(self, collection):
         """
         super(AdvancedQuery, self).__init__(collection)
 
-    def __call__(self, query):
+    def __call__(self, query, attr=None):
```
Comment (ryankwilliams) [inline miqcli/query.py:136]: `attr` parameter is missing from the doc string.

**Review by ryankwilliams — APPROVED** [review]
LGTM. I added 2 comments about missing parameters being documented for future reference.

**Pull request merged**